### PR TITLE
Fix pypi build

### PR DESCRIPTION
**Pull request by tofarr**
This is intended to address https://github.com/All-Hands-AI/OpenHands/issues/3527

As pointed out by a user, our build on pypi does not include agenthub. The package on pypi is pretty much broken due to missing files and dependencies.

## After deploy and a new version is built to pypi, you should be able to run without errors:

* Creating a new project using poetry `poetry new test-open-hands-pypi`
* Add open hands as a dependency: `poetry add openhands-ai`
*  Install dependencies: `poetry install`
* Run a python console: `poetry run python`
* Try to import a core package: `from openhands.core.main import run_controller`
* The import succeeds!

image

### Before the deploy to pypi you need to build and test OpenHands locally
Then specify the location of your local version in your pyproject.toml of your test project:

e.g:
```
cd OpenHands
poetry build
```

Then :
`openhands = { path = "/Users/tofarr/dev/all-hands/OpenHands/dist/openhands-0.8.4.tar.gz" }`
image

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,12 +1,16 @@
 [tool.poetry]
 name = "openhands"
-version = "0.8.3"
+version = "0.8.4"
```
Comment (neubig) [inline pyproject.toml:3]: We probably shouldn't cut a new release until we're pretty sure that we don't have any regressions in agent quality. I think @mamoodi is taking a final look now.

**Review by neubig — CHANGES_REQUESTED** [review]
Looks good to me! But we should probably not cut a release just yet. When we do, probably bumping to 0.9.0 would be good.

**Comment by tobitege** [comment]
@tofarr When you change the version number, please also apply that to the 2 `/frontend/package*.json` files :)

**Comment by neubig** [comment]
Hey @tofarr , we're going to have to change the name of the package to openhands-ai to match the pypi distribution. I changed that in a commit, but that's breaking other things e.g. in the runtime tests, which will have to be fixed.

**Comment by neubig** [comment]
Just for reference, I tested this out locally and it seems to be working if I add:
```
openhands-ai = { git = "https://github.com/All-Hands-AI/openhands.git", rev = "fix-pypi-package" }
```

to `pyprojects.toml`. So once tests are working I think it's good to go!

**Review by neubig — APPROVED** [review]
LGTM now that 0.9.0 is released, lets see if tests pass.

**Pull request merged**